packages/actor-query-operation-filter-sparqlee/lib/ActorQueryOperationFilterSparqlee.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  ActorQueryOperation, ActorQueryOperationTypedMediated, Bindings,
+  IActorQueryOperationOutput,
+  IActorQueryOperationOutputBindings,
+  IActorQueryOperationTypedMediatedArgs,
+} from "@comunica/bus-query-operation";
+import { BindingsStream } from "@comunica/bus-query-operation";
+import { ActionContext, IActorTest } from "@comunica/core";
+import { Algebra } from "sparqlalgebrajs";
+import { AsyncEvaluator, ExpressionError } from "sparqlee";
+
+/**
+ * A comunica Filter Sparqlee Query Operation Actor.
+ */
+export class ActorQueryOperationFilterSparqlee extends ActorQueryOperationTypedMediated<Algebra.Filter> {
+
+  constructor(args: IActorQueryOperationTypedMediatedArgs) {
+    super(args, 'filter');
+  }
+
+  public async testOperation(pattern: Algebra.Filter, context: ActionContext): Promise<IActorTest> {
+    // Will throw error for unsupported operators
+    // Coerce to boolean to check for falsy values
+    return !!new AsyncEvaluator(pattern.expression);
+  }
+
+  public async runOperation(pattern: Algebra.Filter, context: ActionContext)
+    : Promise<IActorQueryOperationOutputBindings> {
+
+    const outputRaw = await this.mediatorQueryOperation.mediate({ operation: pattern.input, context });
+    const output = ActorQueryOperation.getSafeBindings(outputRaw);
+    ActorQueryOperation.validateQueryOutput(output, 'bindings');
+    const { variables, metadata } = output;
+
+    const evaluator = new AsyncEvaluator(pattern.expression);
+    const transform = async (item: Bindings, next: any) => {
+      try {
+        const result = await evaluator.evaluateAsEBV(item);
+        if (result === true) { bindingsStream._push(item); }
```

Comment:
I'm surprised the linter does not complain about this :-) Could you put this on separate lines?

Also, can you add a comment explaining the reason why you should explicitly compare to `true`, and not any thruthy value?

Reply:
I'll put it on separate lines. If the result is not a boolean, something went very wrong (since evaluateAsEBV coerces any term into a boolean/ Effective Boolean Value). Since this API should be stable, I'll replace by a truthy check.